### DISCUSSION TITLE
Running auto3dseg_hello_world.py triggers a multi-processing error on multi-GPU cluster

### DISCUSSION THREAD

**Discussion by dvitaly07** [Project-MONAI/tutorials#1790] [Q&A]
Running auto3dseg_hello_world.py triggers a multi-processing error on multi-GPU cluster, please advise:
![Screenshot from 2024-08-18 11-11-41](https://github.com/user-attachments/assets/c55b20f6-ed9d-4963-b83c-473a9534def1)
![Screenshot from 2024-08-18 11-13-31](https://github.com/user-attachments/assets/76679c30-1f4d-455e-ae23-3590a3e2aa22)

**Comment by KumoLiu** [comment]
Hi @dvitaly07, thanks for your interest here.
The Auto3DSeg modules support multiple GPU since MONAI 1.2. The GPU devices are selected via setting the environment variable CUDA_VISIBLE_DEVICES.
https://github.com/Project-MONAI/tutorials/blob/main/auto3dseg/notebooks/auto_runner.ipynb

**Reply by dvitaly07** [reply]
Hi, unfortunately setting the CUDA_VISIBLE_DEVICES environment variable did not resolve the issue. I am still getting the same exact error message. Please advise.

**Reply by KumoLiu** [reply]
I don't know which notebook do you refer to.
I don't think all the part in this [notebook](https://github.com/Project-MONAI/tutorials/blob/main/auto3dseg/notebooks/auto3dseg_hello_world.ipynb) can support multigpu such as prepare data, you need modify it by yourself.
You can refer to this folder for more details: https://github.com/Project-MONAI/tutorials/tree/main/acceleration/distributed_training

Thanks.

**Comment by dvitaly07** [comment]
Hi, unfortunately setting the CUDA_VISIBLE_DEVICES environment variable did not resolve the issue. I am still getting the same exact error message. Please advise.

    On Monday, August 19, 2024 at 12:04:22 AM EDT, YunLiu ***@***.***> wrote:  
 
 


Hi @dvitaly07, thanks for your interest here.
The Auto3DSeg modules support multiple GPU since MONAI 1.2. The GPU devices are selected via setting the environment variable CUDA_VISIBLE_DEVICES.
https://github.com/Project-MONAI/tutorials/blob/main/auto3dseg/notebooks/auto_runner.ipynb
